src/editors/playgroundController.ts

```diff
@@ -579,7 +572,37 @@ export default class PlaygroundController {
     }
   }
 
-  async _evaluatePlayground(): Promise<boolean> {
+  async evaluateParticipantQuery(text: string): Promise<boolean> {
+    const shouldConfirmRunCopilotCode = vscode.workspace
+      .getConfiguration('mdb')
+      .get('confirmRunCopilotCode');
+
+    if (shouldConfirmRunCopilotCode === true) {
+      const name = this._connectionController.getActiveConnectionName();
+      const confirmRunCopilotCode = await vscode.window.showInformationMessage(
+        `Are you sure you want to run this code generated by the MongoDB participant against ${name}? This confirmation can be disabled in the extension settings.`,
+        { modal: true },
+        'Yes'
+      );
+
+      if (confirmRunCopilotCode !== 'Yes') {
+        return false;
+      }
+    }
+
+    const evaluateResponse: ShellEvaluateResult =
+      await this._evaluateWithCancelModal(text);
+
+    if (!evaluateResponse || !evaluateResponse.result) {
+      return false;
+    }
+
+    await this._openInResultPane(evaluateResponse.result);
+
+    return Promise.resolve(true);
```

Comment:
[nit] I don't believe we need the `Promise.resolve` here since we're already in an async context.

```suggestion
    return true;
```